pkg/cloudprovider/aws/capacity.go

```diff
@@ -16,79 +16,269 @@ package aws
 
 import (
 	"context"
+	"encoding/base64"
+	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/eks/eksiface"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
 	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/utils/log"
+	"go.uber.org/zap"
+)
+
+const (
+	DefaultLaunchTemplateName = "KarpenterDefaultLaunchTemplate"
```

Comment:
Good point. This code probably needs to be factored as part of the reconciliation loop (lazily), because you could imagine wanting one of these for each resource. I'm gonna make a TODO for this.